src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
 		return !getOwnerId().isEmpty();
 	}
 	
+	/**
+	 * Use for XML loading
+	 */
+	public void setNextPlayerSlaveryUpdateTimes(long eventUpdate, long ruleUpdate)
+	{
+		this.timeOfNextEventCheck = eventUpdate;
+		this.timeOfNextRuleCheck = ruleUpdate;
+	}
+	
+	/**
+	 * Use for exceptionally actions that are likely to cause an event to trigger, like resting. Generally just try to pad out any possible time consuming event with those prior to calling it.
+	 * 
+	 * It will delay the event's trigger for five minutes after this one or not delay it at all if it can fit in.
+	 * @param amount
+	 */
+	public void delayEventCheck(long amount)
+	{
+		long projectedDelayTime = Main.game.getMinutesPassed() + amount + GameCharacter.MINIMUM_POST_DELAY_DURATION;
+		if(projectedDelayTime > this.timeOfNextEventCheck)
+		{
+			this.timeOfNextEventCheck = projectedDelayTime;
+		}
+	}
+	
+	/**
+	 * Returns true if player has all rules at least at 0 obedience gain.
+	 */
+	public boolean isPlayerObeyingRules()
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getPerformanceQuality() < 0)
+			{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	/**
+	 * Handles player slavery updates
+	 */
+	public void handlePlayerSlavery(long minutesPassed)
+	{		
+		
+		boolean forcePunishment = false; // If true, forces the player to take punishment. Use for high obedience drops.
+		
+		// Checking rules compliance here
+		if(this.timeOfNextRuleCheck <= minutesPassed)
+		{
+			int totalRules = 0;
+			float obedienceChangePositive = 0;
+			float obedienceChangeNegative = 0;
+			float obedienceChangeTotal = 0;
+			
+			for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+				totalRules++;
```

Comment:
totalRules is simply this.slaveOwnerRules.size() and you don't need an incrementing variable for this